### PR TITLE
new version of audio2text

### PR DESCRIPTION
**Pull request by piotrek6641**

### CODE REVIEW

File: typescript/audio2text-event/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import { ReadableApp } from "@scramjet/types";
+import { Assembly } from "../../audio2text/src/utils/assembly";
+import { PassThrough, Readable } from "stream";
+
+const app: ReadableApp<any> = async function(input: Readable, apiKey: string) {
+    const assembly = new Assembly(apiKey);
+
+    assembly.logger.outputLogStream.pipe(this.logger.inputLogStream);
+    const output = new PassThrough({ defaultEncoding:"utf-8" });
```
Comment (patuwwy) [inline typescript/audio2text-event/src/index.ts:9]:
```suggestion
    const assembly = new Assembly(apiKey);
    const output = new PassThrough({ defaultEncoding:"utf-8" });
    
    assembly.logger.outputLogStream.pipe(this.logger.inputLogStream);

```

File: typescript/audio2text-event/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    let timer;
+    let helpStream: PassThrough = new PassThrough();
+
+    function startTimer() {
+        timer = setTimeout(() => {
+            input.emit("ready");
+        }, 2000);
+    }
```
Comment (patuwwy) [inline typescript/audio2text-event/src/index.ts:17]:
```suggestion
    }

```

File: typescript/audio2text/package.json [hunk 3]
```diff
@@ -14,12 +14,13 @@
...
-    "type":"git",
-    "url":"https://github.com/scramjetorg/platform-samples/tree/main/typescript/audio2text"
+    "type": "git",
+    "url": "https://github.com/scramjetorg/platform-samples/tree/main/typescript/audio2text"
   },
   "author": "piotrek6641",
   "license": "ISC",
   "dependencies": {
+    "@scramjet/obj-logger": "^0.34.3",
     "@scramjet/utility": "^0.33.5",
```
Comment (patuwwy) [inline typescript/audio2text/package.json:24]: please align utility version : ^0.34.3"

File: typescript/audio2text/src/index.ts [hunk 4]
```diff
@@ -7,9 +7,12 @@ const app: ReadableApp<any> = async function(
     ) {
     const assembly = new Assembly(token);
 
-    await assembly.processFile();
+    assembly.logger.outputLogStream.pipe(this.logger.inputLogStream);
 
-    return Promise.resolve("finished");
+    return assembly.processFileFs().then(res => {
+        this.logger.info("Result", res);
+        return res;
```
Comment (patuwwy) [inline typescript/audio2text/src/index.ts:14]:
```suggestion

        return res;
```

File: typescript/audio2text/src/utils/assembly.ts [hunk 5]
```diff
@@ -3,39 +3,42 @@ import axios, { AxiosInstance } from "axios";
 import { defer } from "@scramjet/utility";
 import { readFile } from "fs";
 import path from "path";
+import { ObjLogger } from "@scramjet/obj-logger";
+import { Readable } from "stream";
 
 
 export class Assembly {
     //path to file change song.wav to name of your file
```
Comment (patuwwy) [inline typescript/audio2text/src/utils/assembly.ts:11]: filename could be provided as seq param and passed to `uploadFromFS`

File: typescript/audio2text/src/utils/assembly.ts [hunk 6]
```diff
@@ -3,39 +3,42 @@ import axios, { AxiosInstance } from "axios";
...
-    fileUrl:string = "";
-    status:string = "";
-    transciptId:string = "";
-    token:string;
-    client:AxiosInstance;
+    logger = new ObjLogger("Assembly");
+    fileUrl: string = "";
+    transciptId: string = "";
+    client: AxiosInstance;
     constructor(token:string) {
```
Comment (patuwwy) [inline typescript/audio2text/src/utils/assembly.ts:18]: missing space after :

File: typescript/audio2text/src/utils/assembly.ts [hunk 7]
```diff
@@ -46,30 +49,45 @@ export class Assembly {
...
+            await defer(3000);
+            return this.transcript();
+        }
+
+        return result.text;
+    }
+    async processFileFs(): Promise<string> {
+        await this.uploadFromFs();
+        await this.startTranscript();
+        return this.transcript();
```
Comment (patuwwy) [inline typescript/audio2text/src/utils/assembly.ts:85]:
```suggestion

        return this.transcript();
```

File: typescript/audio2text/src/utils/assembly.ts [hunk 8]
```diff
@@ -46,30 +49,45 @@ export class Assembly {
...
     }
-    async processFile():Promise<void> {
-        await this.upload();
+    async processFile(inputStream: Readable): Promise<string> {
+        await this.upload(inputStream);
         await this.startTranscript();
-        await this.transcript().then(console.log);
+        return this.transcript();
```
Comment (patuwwy) [inline typescript/audio2text/src/utils/assembly.ts:90]:
```suggestion

        return this.transcript();
```

File: typescript/audio2text-event/src/index.ts [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+    });
+    input.on("ready", async () => {
+        helpStream.end();
+        await assembly.processFile(helpStream).then(res => {
+            this.logger.info("Result", res);
+            helpStream = new PassThrough();
+            output.push(res);
+        });
+    });
+    return output;
```
Comment (patuwwy) [inline typescript/audio2text-event/src/index.ts:40]:
```suggestion
    input
        .once("data", () => {
            startTimer();
        })
        .on("data", (chunk) => {
            helpStream.push(chunk);
            clearTimeout(timer);
            startTimer();
        })
        .on("ready", async () => {
            helpStream.end();
            await assembly.processFile(helpStream).then(res => {
                this.logger.info("Result", res);
                helpStream = new PassThrough();
                output.push(res);
            });
        });

    this.on("pause", () => {
        input.pause();
    });

    this.on("resume", () => {
        input.resume();
    });

    return output;
```

File: typescript/audio2text-event/src/index.ts [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+        clearTimeout(timer);
+        startTimer();
+    });
+    this.on("pause", () => {
+        input.pause();
+    });
+    this.on("resume", () => {
+        input.resume();
+    });
+    input.on("ready", async () => {
```
Comment (patuwwy) [inline typescript/audio2text-event/src/index.ts:32]: `this.on("ready"... ` ?

**Review by alicja-gruzdz — APPROVED** [review]

**Pull request merged**